myriad-scheduler/src/main/java/com/ebay/myriad/scheduler/event/handlers/ResourceOffersEventHandler.java

```diff
@@ -162,30 +164,31 @@ private boolean matches(Offer offer, NMProfile profile) {
     checkResource(mem < 0, "mem");
     checkResource(ports < 0, "port");
 
-    return checkAggregates(offer, profile, ports, cpus, mem);
+    return checkAggregates(offer, taskToLaunch, ports, cpus, mem);
   }
 
+    private boolean checkAggregates(Offer offer, NodeTask taskToLaunch, int ports, double cpus, double mem) {
+        Map<String, String> requestAttributes = new HashMap<>();
+        final double aggrCpu = taskUtils.getAuxTaskCpus(taskToLaunch.getProfile(), taskToLaunch.getTaskPrefix());
+        final double aggrMem = taskUtils.getAuxTaskMemory(taskToLaunch.getProfile(), taskToLaunch.getTaskPrefix());
+        if (aggrCpu <= cpus
+                && aggrMem <= mem
+                && SchedulerUtils.isMatchSlaveAttributes(offer, requestAttributes)
+                && NMPorts.expectedNumPorts() <= ports) {
+            return true;
```

Comment:
Good catch - will look at it.
JHS should not be using random web port since it is the one clients connect to, so preferably it should be allocated in the list of ports allocated when RM is defined in Marathon